airbyte-ci/connectors/pipelines/pipelines/builds/python_connectors.py

```diff
@@ -15,13 +17,100 @@ class BuildConnectorImage(BuildConnectorImageBase):
     A spec command is run on the container to validate it was built successfully.
     """
 
+    DEFAULT_ENTRYPOINT = ["python", "/airbyte/integration_code/main.py"]
+    PATH_TO_INTEGRATION_CODE = "/airbyte/integration_code"
+
+    @property
+    def _build_connector_function(self):
+        if (
+            "connectorBuildOptions" in self.context.connector.metadata
+            and "baseImage" in self.context.connector.metadata["connectorBuildOptions"]
+        ):
+            return self._build_from_base_image
+        else:
+            return self._build_from_dockerfile
+
     async def _run(self) -> StepResult:
-        connector = await with_airbyte_python_connector(self.context, self.build_platform)
+        connector: Container = await self._build_connector_function()
+        connector = await apply_python_development_overrides(self.context, connector)
         try:
             return await self.get_step_result(connector.with_exec(["spec"]))
         except QueryError as e:
             return StepResult(self, StepStatus.FAILURE, stderr=str(e))
 
+    def _get_base_container(self) -> Container:
+        base_image_name = self.context.connector.metadata["connectorBuildOptions"]["baseImage"]
+        self.logger.info(f"Building connector from base image {base_image_name}")
+        return self.dagger_client.container(platform=self.build_platform).from_(base_image_name)
+
+    async def _provision_builder_container(self, base_container: Container) -> Container:
```

Comment:
📚 provision makes me think create a worker, or ask external service for resource.

could we use init, create, or build instead?